workflows/cloudify_system_workflows/snapshots/postgres.py

```diff
@@ -200,30 +203,76 @@ def _get_admin_user_update_query(self):
         return (base_query.format(username, password),
                 base_query.format('*'*8, '*'*8))
 
+    @staticmethod
+    def _find_reporter_role(roles_mapping, reporter_id):
+        reporter_role = [role_id for user_id, role_id in roles_mapping
+                         if user_id == reporter_id]
+        if reporter_role:
+            return role_id
```

Comment:
let's not do hacks like this (also it won't work in py3)

`roles_mapping` isn't really a mapping right now, so let's make it a mapping by doing `return dict(response['all'])` in `_get_status_reporters_roles`. Then this method will be just `try: return roles_mapping[reporter_id]; except KeyError: ...`